h2o-bindings/bin/custom/R/gen_stackedensemble.py

```diff
@@ -35,6 +35,32 @@ def update_param(name, param):
     base_models[[i]] <- h2o.keyof(base_models[[i]])
   }
 }
+
+# Warn user if all base models use same weights_column but user didn't specify any for SE
+if (missing(weights_column)) {
+    capture.output({ # don't print error stack traces from Java in case base_model_key is a grid...
+      weights_columns <- lapply(base_models, function (base_model_key) {
+        tryCatch({
+            tryCatch(h2o.getModel(base_model_key)@allparameters$weights_column$column_name,
+                     error=function (e) {
+                       unlist(lapply(h2o.getGrid(base_model_key)@model_ids, function (bmk) {
+                          h2o.getModel(bmk)@allparameters$weights_column$column_name
+                        }))
+                     })
+        }, error=function (e)
+          stop(paste0("Unsupported type of base model \\\"", base_model_key, "\\\". Should be either a model or a grid.")))
+      })
+    })
+    reference_weights_column <- weights_columns[[1]]
+    if (!is.null(reference_weights_column)) {
+      if (all(sapply(weights_columns, function(wc) !is.null(wc) && wc == reference_weights_column))) {
+        warning(paste0("All base models use weights_column=\\\"", reference_weights_column,
+                       "\\\" but Stacked Ensemble does not. If you want to use the same ",
+                       "weights_column for the meta learner, please specify it as an argument ",
+                       "in the h2o.stackedEnsemble call."))
+      }
+    }
+}
```

Comment:
Non-trivial validation logic should be done in backend instead of being replicated in each client.
Especially when like here, it requires loading all base modelson the client.
Also, this kind of validation won't work in Flow, and would need to be replicated in SparklingWater as well.

Instead, you should be able to use `ModelBuilder.warn` when doing the validation in `StackedEnsemble`.
This populates `  public ValidationMessageV3 messages[];` in `ModelBuilderV3`, which is then accessible through REST API.
For example, in Python, those messages should be accessible in `estimator_base.py`, method `_train`:
```
        model_builder_json = h2o.api("POST /%d/ModelBuilders/%s" % (rest_ver, self.algo), data=parms)
```
I've often been frustrated that we don't have this mechanism for AutoML, but it does exist for other algos, we should just capture the messages there and emit the warnings on the client.